response_operations_ui/common/redis_cache.py

```diff
@@ -0,0 +1,74 @@
+import json
+import logging
+
+from flask import current_app
+from redis.exceptions import RedisError
+from structlog import wrap_logger
+
+from response_operations_ui import redis
+from response_operations_ui.controllers.cir_controller import get_cir_metadata
+from response_operations_ui.controllers.survey_controllers import (
+    get_survey_by_shortname,
+)
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class RedisCache:
+    SURVEY_EXPIRY = 600  # 10 mins
+
+    def get_cir_metadata(self, survey_ref, formtype):
+        """
+        Gets the cir_metadata from redis or the cir service
+
+        :param short_name: str: the qualifying part of the redis key (response-operations-ui:survey:<SURVEY_REF>:<FORMTYPE>)
+        :return: Result from either the cache or the CIR service
+        """
+        redis_key = f"response-operations-ui:cir:{survey_ref}:{formtype}"
+        try:
+            result = current_app.redis.get(redis_key)
+        except RedisError:
+            logger.error("Error getting value from cache, please investigate", key=redis_key, exc_info=True)
+            result = None
+
+        if not result:
+            logger.info("Key not in cache, getting value from CIR service", key=redis_key)
+            result = get_cir_metadata(survey_ref, formtype)
+            current_app.redis.set(redis_key, json.dumps(result), self.SURVEY_EXPIRY)
```

Comment:
If the object is retrieved from the API by the controller, it is added to the cache for subsequent retrieval